### DISCUSSION TITLE
Is Vulkan SC supported/planned to be supported ?

### DISCUSSION THREAD

**Discussion by filipwasil** [KhronosGroup/MoltenVK#1841] [General]
Are there plans to have an out of the box support for Safety Critical Vulkan version ? ( aka Vulkan SC )
As Vulkan 1.2 is available, the Vulkan SC should be "just" a subset of the regular API.

Couldn't find any info on that.

**Comment by billhollings** [comment]
We do not have any current plans to specifically address Vulkan SC.

But interesting request. What is your requirement for deploying Vulkan SC on macOS or iOS?

**Reply by filipwasil** [reply]
Hi Bill and Thank You for this insight.
I would like to be able to run a Vulkan SC app which i could later run on an actual Safety Critical hardware.
There is no actual need on my end to target iOS or macOS directly.
Just for prototyping.

**Reply by billhollings** [reply]
Thanks for the feedback.

TBH...it's going to be low on our priority list, because we have a substantial backlog just on continuing to chase non-SC Vulkan.

But, based on your original post, since you seem to know what's needed, we would be very happy to get a PR from you creating a Vulkan SC emulator. We wouldn't be able to advertise it as SC, because presumably that requires some significant testing and certification, but from an API emulation perspective, feel free to submit something.

**Reply by filipwasil** [reply]
Yeah, I did some research and looks like some of these features are not really a "subset" of Vulkan 1.2, but a new ones:

There is this blog post by Khronos
https://www.khronos.org/blog/vulkan-sc-overview

and I made a quick list of the structs that are in fact missing in Vulkan 1.2:

•	VkPhysicalDeviceVulkanSC10Features 
•	VkPhysicalDeviceVulkanSC10Properties
 •	VkDeviceObjectReservationCreateInfo 
•	VkPipelineCacheHeaderVersionSafetyCriticalOne
 •	VkPipelineCacheSafetyCriticalIndexEntry 
•	VkPipelineOfflineCreateInfo 
•	VkCommandPoolMemoryReservationCreateInfo 
•	vkGetCommandPoolMemoryConsumption 
•	VkFaultCallbackInfo
 •	VkFaultData 

Maybe I will put together something that could be a good start for further development.
